### PR TITLE
ARROW-10871: [Julia][CI] Setup Julia testing via Github Actions

### PR DESCRIPTION
**Pull request by quinnj**

### CODE REVIEW

File: .github/workflows/julia.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# specific language governing permissions and limitations
+# under the License.
+
+name: Julia
+on:
+  push:
+    paths:
+      - '.github/workflows/julia.yml'
+      - 'ci/docker/*_julia.dockerfile'
+      - 'ci/scripts/julia_*'
```
Comment (nealrichardson) [inline .github/workflows/julia.yml:24]: There are no Julia files in the ci/ directory, are there? I think you can delete these and just trigger on changes to this workflow file and changes to the julia directory.

Comment (quinnj) [reply .github/workflows/julia.yml]: Yeah, I just copied what other languages were doing; updated.

File: .github/workflows/julia.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  pull_request:
+    paths:
+      - '.github/workflows/julia.yml'
+      - 'ci/docker/*_julia.dockerfile'
+      - 'ci/docker/**'
+      - 'julia/**'
+
+jobs:
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
```
Comment (kou) [inline .github/workflows/julia.yml:34]: We use `${CPU} ${PLATFORM} ${LANGUAGE} ${OPTIONAL_NOTE}` format in our jobs.
For example:

  * `AMD64 Ubuntu 18.04 C++ ASAN UBSAN` where `${CPU}=AMD64`, `${PLATFORM}=Ubuntu 18.04`, `${LANGUAGE}=C++` and `${OPTIONAL_NOTE}=ASAN UBSAN`
  * `AMD64 Conda Python 3.8 Without Pandas` where `${CPU}=AMD64`, `${PLATFORM=Conda`, `${LANGUAGE}=Python 3.8` and `${OPTIONAL_NOTE}=WIthout Pandas`

Can we follow the style?

File: .github/workflows/julia.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Julia
+on:
+  push:
+    paths:
+      - '.github/workflows/julia.yml'
+      - 'ci/docker/*_julia.dockerfile'
```
Comment (kou) [inline .github/workflows/julia.yml:23]:
```suggestion
```

File: .github/workflows/julia.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      - 'julia/**'
+  pull_request:
+    paths:
+      - '.github/workflows/julia.yml'
+      - 'ci/docker/*_julia.dockerfile'
+      - 'ci/docker/**'
```
Comment (kou) [inline .github/workflows/julia.yml:29]:
```suggestion
```

File: .github/workflows/julia.yml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      fail-fast: false
+      matrix:
+        version:
+          - '1.3'
+          - '1' # automatically expands to the latest stable 1.x release of Julia
+          - 'nightly'
+        os:
+          - ubuntu-latest
+        arch:
+          - x64
```
Comment (kou) [inline .github/workflows/julia.yml:48]: If we don't have a plan to add more `arch` (`x86`) in near future, how about removing this matrix item?

File: .github/workflows/julia.yml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - '1.3'
+          - '1' # automatically expands to the latest stable 1.x release of Julia
+          - 'nightly'
+        os:
+          - ubuntu-latest
```
Comment (kou) [inline .github/workflows/julia.yml:46]: Is it enough to run this job only on Ubuntu?
Do we need to run this job on macOS and Windows too?

File: .github/workflows/julia.yml (hunk 2 above)
Comment (quinnj) [reply .github/workflows/julia.yml]: Updated

File: .github/workflows/julia.yml (hunk 4 above)
Comment (quinnj) [reply .github/workflows/julia.yml]: Updated

File: .github/workflows/julia.yml (hunk 5 above)
Comment (quinnj) [reply .github/workflows/julia.yml]: Removed

File: .github/workflows/julia.yml (hunk 6 above)
Comment (quinnj) [reply .github/workflows/julia.yml:43]: I've added windows too; we don't expect there to be any platform differences between linux/osx, so we stopped testing that a while ago to try and be better citizens around free github action minutes. We can test windows too, but again, the julia binary works out of the box well across platforms and we're not doing anything in the Arrow.jl code that is very platform-specific at all.

Comment (nealrichardson) [reply .github/workflows/julia.yml:43]: IMO this is fine for now, but if we become concerned about the number of GHA jobs/minutes, we can move some of these builds to nightly/on-demand. Like, run Windows w/Julia current release and Linux w/Julia nightly on every commit, and with those 2 jobs you get most of the coverage you care about; then move the other parts of the build matrix to nightly jobs.

Comment (kou) [reply .github/workflows/julia.yml:43]: @quinnj I see.

@nealrichardson +1

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**